xbmc/video/dialogs/GUIDialogVideoVersion.cpp

```diff
@@ -0,0 +1,1008 @@
+  {
+    std::string strVideoVersion;
+    std::string strVideoTitle;
+    int idFile;
+    int idMedia;
+    MediaType itemMediaType;
+    VideoVersionItemType versionItemType;
+
+    int idVideoVersion = m_database.GetVideoVersionInfo(path, idFile, strVideoVersion, idMedia,
+                                                        itemMediaType, versionItemType);
+    if (idVideoVersion != -1)
+    {
+      CFileItemList versions;
+      m_database.GetVideoVersions(itemType, dbId, versions);
+      if (std::any_of(versions.begin(), versions.end(),
+                      [idFile](const std::shared_ptr<CFileItem>& version)
+                      { return version->GetVideoInfoTag()->m_iDbId == idFile; }))
+      {
+        CGUIDialogOK::ShowAndGetInput(
+            title, StringUtils::Format(g_localizeStrings.Get(40016), strVideoVersion,
+                                       CMediaTypes::GetLocalization(mediaType)));
+        return;
+      }
+
+      if (itemMediaType == MediaTypeMovie)
+      {
+        strVideoTitle = m_database.GetMovieTitle(idMedia);
+      }
+      else
+        return;
+
+      if (!CGUIDialogYesNo::ShowAndGetInput(
+              title, StringUtils::Format(g_localizeStrings.Get(40017), strVideoVersion,
+                                         CMediaTypes::GetLocalization(mediaType), strVideoTitle,
+                                         CMediaTypes::GetLocalization(mediaType))))
+      {
+        return;
+      }
+
+      if (m_database.IsDefaultVideoVersion(idFile))
+      {
+        CFileItemList list;
+        m_database.GetVideoVersions(itemType, idMedia, list);
+
+        if (list.Size() > 1)
+        {
+          CGUIDialogOK::ShowAndGetInput(
+              title, StringUtils::Format(g_localizeStrings.Get(40019), strVideoVersion,
+                                         CMediaTypes::GetLocalization(mediaType), strVideoTitle));
+          return;
+        }
+        else
+        {
+          if (itemMediaType == MediaTypeMovie)
+          {
+            m_database.DeleteMovie(idMedia);
+          }
+          else
+            return;
+        }
+      }
+      else
+        m_database.RemoveVideoVersion(idFile);
+    }
+
+    CFileItem item(path, false);
+
+    if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+            CSettings::SETTING_MYVIDEOS_EXTRACTFLAGS))
+    {
+      CDVDFileInfo::GetFileStreamDetails(&item);
+      CLog::Log(LOGDEBUG, "VideoVersion: Extracted filestream details from video file {}",
+                CURL::GetRedacted(item.GetPath()));
+    }
+
+    if (primary)
+    {
+      const int idVideoVersion = SelectVideoVersion(m_videoItem);
+      if (idVideoVersion != -1)
+      {
+        m_database.AddPrimaryVideoVersion(itemType, dbId, idVideoVersion, item);
+      }
+    }
+    else
+    {
+      std::string strVideoVersion =
+          CGUIDialogVideoVersion::GenerateExtrasVideoVersion(URIUtils::GetFileName(path));
+      int idVideoVersion =
```

Comment:
`const int`

(Maybe you run once through all you newly added code to check and fix this everywhere? I'm getting a bit exhausted having to say the same thing again and again, tbh.)